File: tournaments/ballsDistribution/ballsDistribution.cs
int ballsDistribution(int colors, int ballsPerColor, int boxSize) {
    var result = 0;
    var a = 0;
    var b = boxSize;

    for (var i = 0; colors > i; ++i) {
        var c = a;
        for (var j = 0; ballsPerColor > j; ++j) {
            --b;
            if (0 == b) {
                b = boxSize;
                ++a;
            }
        }
        if (c < a && (boxSize > b || c + 1 < a)) {
            ++result;
        }
    }

    return result;
}

File: tournaments/ballsDistribution/ballsDistribution.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub ballsDistribution {
    my ($colors, $ballsPerColor, $boxSize) = @_;

    my $result = 0;
    my $c = 0;
    my $d = $boxSize;

    for (my $i = 0; $colors > $i; ++$i) {
        my $e = $c;
        for (my $j = 0; $ballsPerColor > $j; ++$j) {
            --$d;
            if (!$d) {
                $d = $boxSize;
                ++$c;
            }
        }
        if ($e < $c && ($boxSize > $d || $e + 1 < $c)) {
            ++$result;
        }
    }

    return $result;
}
